core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala

```diff
@@ -723,20 +733,97 @@ object IfConditionalScriptPubKey
     fromAsm(asm)
   }
 
-  def isIfConditionalScriptPubKey(asm: Seq[ScriptToken]): Boolean = {
-    ConditionalScriptPubKey
+  def isNonStandardIfConditionalScriptPubKey(asm: Seq[ScriptToken]): Boolean = {
+    val validIf = ConditionalScriptPubKey
       .isConditionalScriptPubKeyWithElseIndex(asm, OP_IF)
       ._1
+
+    val isMultiSigWithTimeout = MultiSignatureWithTimeoutScriptPubKey
+      .isMultiSignatureWithTimeoutScriptPubKey(asm)
+
+    validIf && !isMultiSigWithTimeout
+  }
+}
+
+sealed trait MultiSignatureWithTimeoutScriptPubKey
+    extends IfConditionalScriptPubKey {
+  require(
+    MultiSignatureScriptPubKey.isMultiSignatureScriptPubKey(super.firstSPK.asm),
+    "True case must be MultiSignatureSPK")
+  require(CLTVScriptPubKey.isCLTVScriptPubKey(super.secondSPK.asm),
+          "False case must be CLTVSPK")
+
+  override def firstSPK: MultiSignatureScriptPubKey = {
+    MultiSignatureScriptPubKey.fromAsm(asm.slice(1, opElseIndex))
+  }
+
+  override def secondSPK: CLTVScriptPubKey = {
```

Comment:
Any reason for using `CLTVScriptPubKey` rather than a more generic `LockTimeScriptPubKey`?

Reply:
I'm open to doing that in the future but I wanted to narrow the scope to what I'll actually use on the first iteration of DLCs for now so as not to bloat, I believe making it `LockTime` does add some complexity IIRC